block_acclaim.php

```diff
@@ -53,18 +53,19 @@ public function has_config() {
      * Display specialized text for the course block (the selected badge name).
      */
     public function specialization() {
-        if (isset($this->config)) {
-            global $COURSE;
-            $course_id = $COURSE->id;
-
-            $badge_name = $this->acclaim->get_course_info($course_id, 'badgename');
+        if(!isset($this->config)){
+            $this->config = new stdClass();
+        }
+        global $COURSE;
+        $course_id = $COURSE->id;
 
-            if ($badge_name == '') {
-                $badge_name = 'No Badge Selected';
-            }
+        $badge_name = $this->acclaim->get_course_info($course_id, 'badgename');
 
-            $this->config->text = $badge_name;
+        if ($badge_name == '') {
+            $badge_name = 'No Badge Selected';
```

Comment:
This test is fine, but I'd use a falsey check instead (`if (!$badge_name)`) to cover the extremely unlikely case that the 'badgename' field isn't defined at all (which causes `get_course_info` to return null). Up to you.

Reply:
We have a NOT NULL db restraint on badge_name for `block_acclaim_courses` but to guard against future db changes we can add a null check here 👍